### DISCUSSION TITLE
25. July 2023

### DISCUSSION THREAD

**Discussion by dimakuv** [gramineproject/gramine#1466] [Meeting notes]
### Agenda

*(please write your proposed agenda items in comments under this discussion)*

- [x] GSC license (LGPL vs BSD vs ELSE), see https://github.com/gramineproject/gsc/issues/161#issuecomment-1645746933
- [x] Don: post an acknowledgment for the NSF grant on the Gramine page
- [x] Don: budget for CI purchases

### Don: post an acknowledgment for the NSF grant on the Gramine page

NSF would like to see an acknowledgment note on the web page.

Don: who else should we acknowledge?
- Woju: we already thank the Fosshost hosting on gramineproject.io.
- Woju: there is already the "commitment behind Gramine" list, we could add there.
- Benny: by the way, there is an old Intel logo on the web page, we need to replace it.

UPDATE: Woju created a PR with these fixes: https://github.com/gramineproject/gramineproject.io/pull/7

### Don: budget for CI purchases

- Woju: we would like to migrate to the Zuul CI framework. And we can accommodate new machines there.
- Michal: I would like to add an SPR (Sapphire Rapids) machine.
- Don: this budget doesn't need to be spent this year.
- Benny's side note: the list of Intel CPUs that support SGX: https://www.intel.com/content/www/us/en/architecture-and-technology/software-guard-extensions-processors.html

### Dmitrii: GSC license (LGPL vs BSD vs ELSE)

- Michal: the default is a very restrictive license (if there is no license mentioned at all in the repo).
- Woju: We have the SPDX licensing lines in the Python scripts, which makes the license of GSC pretty clear: it's LGPL v3. We don't have the license on the Dockerfile templates, but this may be considered not "creative" enough to have a license at all.
- Michal: GSC is a small tool of several hundred lines of Python code; the situation is different from those described about MongoDB at https://github.com/gramineproject/gsc/issues/161#issuecomment-1642441308.
- Woju: there is a proposal to provide a Dockerfile template to GSC as an argument, and this external template is considered a separate body of work.
    - Michal: this is making life harder to users: they will need to work with templates, with all their limitations.

[ Michal and Woju agreed in the end to change the license to BSD 3-clause. ]

**Comment by donporter** [comment]
I would like to post an acknowledgment for the recent NSF grant somewhere on the Gramine page - just a note to the effect that Grant X (+link) is supporting ongoing development of Gramine, as well as the following required disclaimer:  

 Disclaimer: Any opinions, findings, and conclusions or recommendations expressed in this material are ours alone, and do not necessarily reflect the views of our sponsors. All product names, logos, and brands are property of their respective owners. All company, product and service names used in this website are for identification purposes only. Use of these names, logos, and brands does not imply endorsement.

Where is the right place to do this?  Perhaps as a page under "Resources"?  

I suspect this also raises the issue of other organizations that we should thank being noted somewhere in the docs or on the webpage.

**Comment by donporter** [comment]
Another topic: I have some budget for CI purchases in this grant.  What do we currently need?  I'm guessing more ICX machines?  Anything else?
